web/src/engine/websites/ReaperScans.ts

```diff
@@ -45,40 +50,50 @@ export default class extends DecoratableMangaScraper {
     }
 
     public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
-        const slug = url.match(/\/series\/([^/]+)/)[1];
+        const slug = url.split('/').pop();
         const manga = await this.interactionTaskPool.Add(async () => FetchJSON<APIManga>(new Request(new URL(`/series/${slug}`, this.apiUrl))), Priority.Normal);
         return new Manga(this, provider, manga.id.toString(), manga.title.trim());
     }
 
     public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
-        const mangaList = [];
-        for (let page = 1, run = true; run; page += 1) {
-            const mangas = await this.interactionTaskPool.Add(async () => this.GetMangasFromPage(page, provider), Priority.Normal);
+        const mangaList: Manga[] = [];
+        const uri = new URL(`/query`, this.apiUrl);
+        uri.searchParams.set('perPage', '100');
+        uri.searchParams.set('series_type', 'Comic');
+        uri.searchParams.set('adult', 'true');
+        for (let page = 1, run = true; run; page++) {
+            const mangas = await this.interactionTaskPool.Add(async () => this.GetMangasFromPage(uri, page, provider), Priority.Normal);
             mangas.length > 0 ? mangaList.push(...mangas) : run = false;
         }
         return mangaList;
     }
 
-    private async GetMangasFromPage(page: number, provider: MangaPlugin): Promise<Manga[]> {
-        const { data } = await FetchJSON<APIResult<APIManga[]>>(new Request(new URL(`/query?page=${page}&perPage=100&series_type=Comic&adult=true`, this.apiUrl)));
+    private async GetMangasFromPage(uri: URL, page: number, provider: MangaPlugin): Promise<Manga[]> {
+        uri.searchParams.set('page', page.toString());
+        const { data } = await FetchJSON<APIResult<APIManga[]>>(new Request(uri));
         return data.map(item => new Manga(this, provider, item.id.toString(), item.title.trim()));
     }
 
     public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
-        const chapterList = [];
+        const chapterList: Chapter[] = [];
+        const uri = new URL(`/chapter/query`, this.apiUrl);
+        uri.searchParams.set('perPage', '100');
+        uri.searchParams.set('series_id', manga.Identifier);
         for (let page = 1, run = true; run; page++) {
-            const chapters = await this.interactionTaskPool.Add(async () => this.GetChaptersFromPage(manga, page), Priority.Normal);
+            const chapters = await this.interactionTaskPool.Add(async () => this.GetChaptersFromPage(uri, manga, page), Priority.Normal);
```

Comment:
Chapter loading may use a higher priority than syncing the manga list, otherwise a user will have to wait until the manga list sync is finished before the clicked chapter can be read.